File: src/domain/entities/Budget.js
export class Budget {
  constructor({ id, cat, name, icon, limit, color, notes }) {
    this.id = id || Date.now();
    this.cat = cat;
    this.name = name || cat;
    this.icon = icon || '📦';
    this.limit = Number(limit);
    this.color = color || '#8a94a6';
    this.notes = notes || '';
  }

  isValid() {
    return this.name && this.name.trim().length > 0 && !isNaN(this.limit) && this.limit > 0;
  }
}

File: mobile/lib/models/budget.dart
class AppBudget {
  final int id;
  final String cat;
  final String name;
  final String icon;
  final double limit;
  final String color;
  final String? notes;

  AppBudget({required this.id, required this.cat, required this.name, required this.icon, required this.limit, required this.color, this.notes});

  factory AppBudget.fromJson(Map<String, dynamic> j) => AppBudget(
    id: j['id'] as int,
    cat: j['cat'] as String,
    name: j['name'] as String,
    icon: j['icon'] as String? ?? '📦',
    limit: (j['limit'] as num).toDouble(),
    color: j['color'] as String,
    notes: j['notes'] as String?,
  );

  Map<String, dynamic> toJson() => {'cat': cat, 'name': name, 'icon': icon, 'limit': limit, 'color': color, 'notes': notes ?? ''};
}
